File: 206.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr, n, p = head, None, None 
        while curr: 
           n = curr.next 
           curr.next = p
           p = curr 
           curr = n
        return p

File: 206.cpp
#include <algorithm>
#include <deque>
#include <iostream>
#include <map>
#include <set>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
 	ListNode* curr = head; 
 	ListNode* next = NULL; 
 	ListNode* next = NULL; 
	while (curr){
		next = curr->next; 
		curr->next = prev; 
		prev = curr; 
		curr = next; 
	}
	return prev; 
    }
};
int main() {}
